lib/rouge/lexers/xojo.rb

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Xojo < RegexLexer
+      title "Xojo"
+      desc "Xojo"
+      tag 'xojo'
+      aliases 'realbasic'
+      filenames '*.xojo_code', '*.xojo_window', '*.xojo_toolbar', '*.xojo_menu'
+
+        keywords = %w(
+          addhandler aggregates array asc assigns attributes begin break
+          byref byval call case catch class const continue char ctype declare
+          delegate dim do downto each else elseif end enum event exception
+          exit extends false finally for function global goto if
+          implements inherits interface lib loop mod module
+          new next nil object of optional paramarray
+          private property protected public raise raiseevent rect redim
+          removehandler return select shared soft static step sub super
+          then to true try until using uend uhile
+        )
+
+        keywords_type = %w(
+          boolean cfstringref cgfloat cstring curency date double int8 int16
+          int32 int64 integer ostype pstring ptr short single
+          single string structure variant uinteger uint8 uint16 uint32 uint64
+          ushort windowptr wstring
+        )
+
+        operator_words = %w(
+          addressof and as in is isa mod not or xor
+        )
+
+      state :whitespace do
+        rule /\s+/, Text
+        rule /rem\b.*?$/i, Comment::Single
+        rule /\/\/.*$/, Comment::Single
+        rule /\#tag Note.*\#tag EndNote/m, Comment::Doc
+        rule /".*"/, Literal::String
+      end
+      
+    
+      state :root do
+        mixin :whitespace
+        rule /\s*[#].*$/x, Comment::Preproc
+        rule /[(){}!#,:]/, Punctuation
+        rule /\b(?:#{keywords.join('|')})\b/i, Keyword
+        rule /\b(?:#{keywords_type.join('|')})\b/i, Keyword::Declaration
+        rule /\b(?:#{operator_words.join('|')})\b/i, Operator
+       
+        rule(
+          %r(<=|>=|<>|[=><\+\-\*\/\\]),
+          Operator
+        )
+
+        rule /[+-]?(\d+\.\d*|\d*\.\d+)(f[+-]?\d+)?/i, Literal::Number::Float
```

Comment:
Two points:

1. This rule will match `5.`. The period should only be part of the number if there is a number immediately following it, right?

2. How does the `f` syntax work? I didn't see that in the visual sample.

Reply:
1. Fixed
2. That is a copy/paste error... fixed in latest commit

Reply:
Sorry, I misread that. It should allow 5. as a valid float value.  .5 is also allowed. Updated lexer and sample in the latest commit.